tools/azure-sdk-tools/packaging_tools/change_log.py

```diff
@@ -6,26 +6,31 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _build_md(content, title, buffer):
+    buffer.append(title)
+    buffer.append("")
+    for item in content:
+        buffer.append("  - " + item)
+    buffer.append("")
+
+
 class ChangeLog:
     def __init__(self, old_report, new_report):
         self.features = []
         self.breaking_changes = []
+        self.optional_features = []
         self._old_report = old_report
         self._new_report = new_report
 
     def build_md(self):
         buffer = []
         if self.features:
-            buffer.append("**Features**")
-            buffer.append("")
-            for feature in self.features:
-                buffer.append("  - " + feature)
-            buffer.append("")
+            _build_md(self.features, "**Features**", buffer)
         if self.breaking_changes:
-            buffer.append("**Breaking changes**")
-            buffer.append("")
-            for breaking_change in self.breaking_changes:
-                buffer.append("  - " + breaking_change)
+            _build_md(self.breaking_changes, "**Breaking changes**", buffer)
+        if len(buffer) == 0 and self.optional_features:
```

Comment:
That's not the most logical test to do here, instead of checking if the buffer is empty, it seems better designed to check if we found breaking changes or features:

```python
if not (self.features or self.breaking_changes) and self.optional_features:
```

Reply:
The logical is actually equal, since `buffer` will be added something when `features` or `breaking_changes` exist. But it is ok to change it.